contracts/modules/cover/Quotation.sol

```diff
@@ -69,44 +69,42 @@ contract Quotation is Iupgradable {
   }
 
   /**
-   * @dev Expires a cover after a set period of time.
-   * Changes the status of the Cover and reduces the current
-   * sum assured of all areas in which the quotation lies
-   * Unlocks the CN tokens of the cover. Updates the Total Sum Assured value.
-   * @param _cid Cover Id.
+   * @dev Expires a cover after a set period of time and changes the status of the cover
+   * @dev Reduces the total and contract sum assured
+   * @param coverId Cover Id.
    */
-  function expireCover(uint _cid) public {
-    require(checkCoverExpired(_cid) && qd.getCoverStatusNo(_cid) != uint(QuotationData.CoverStatus.CoverExpired));
+  function expireCover(uint coverId) external {
 
-    tf.unlockCN(_cid);
-    bytes4 curr;
-    address scAddress;
-    uint sumAssured;
-    (,, scAddress, curr, sumAssured,) = qd.getCoverDetailsByCoverID1(_cid);
-    if (qd.getCoverStatusNo(_cid) != uint(QuotationData.CoverStatus.ClaimAccepted))
-      _removeSAFromCSA(_cid, sumAssured);
-    qd.changeCoverStatusNo(_cid, uint8(QuotationData.CoverStatus.CoverExpired));
-  }
+    uint expirationDate = qd.getValidityOfCover(coverId);
+    require(expirationDate < now, "Quotation: cover is not due to expire");
 
-  /**
-   * @dev Checks if a cover should get expired/closed or not.
-   * @param _cid Cover Index.
-   * @return expire true if the Cover's time has expired, false otherwise.
-   */
-  function checkCoverExpired(uint _cid) public view returns (bool expire) {
+    uint coverStatus = qd.getCoverStatusNo(coverId);
+    require(coverStatus != uint(QuotationData.CoverStatus.CoverExpired), "Quotation: cover already expired");
 
-    expire = qd.getValidityOfCover(_cid) < uint64(now);
+    if (coverStatus != uint(QuotationData.CoverStatus.ClaimAccepted)) {
+      (,, address contractAddress, bytes4 currency, uint amount,) = qd.getCoverDetailsByCoverID1(coverId);
+      qd.subFromTotalSumAssured(currency, amount);
+      qd.subFromTotalSumAssuredSC(contractAddress, currency, amount);
+    }
 
+    qd.changeCoverStatusNo(coverId, uint8(QuotationData.CoverStatus.CoverExpired));
```

Comment:
This overwrites the `ClaimAccepted` status, make sure we don't allow new claims to be submitted.